src/core/reference/include/openvino/reference/string_tensor_pack.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+#include "openvino/core/shape.hpp"
+
+namespace ov {
+namespace reference {
+template <typename T_idx>
+void string_tensor_pack(const T_idx* begins,
+                        const T_idx* ends,
+                        const uint8_t* symbols,
+                        std::string* out,
+                        const size_t string_count) {
+    const char* chars = reinterpret_cast<const char*>(symbols);
+    for (int64_t i = 0; i < string_count; ++i) {
```

Comment:
To fix: 
`error: comparison of integer expressions of different signedness: ‘int64_t’ {aka ‘long int’} and ‘const size_t’ {aka ‘const long unsigned int’} [-Werror=sign-compare]`


```suggestion
    for (size_t i = 0; i < string_count; ++i) {
```